regression/contracts/frees-clause-while-loop-pass/main.c

```diff
@@ -0,0 +1,20 @@
+#include <stdlib.h>
+int main()
+{
+  size_t size = 10;
+  char *arr = malloc(size);
+  size_t i = 0;
+  while(i <= size)
+    // clang-format off
+  __CPROVER_assigns(i, arr[2], __CPROVER_POINTER_OBJECT(arr))
+  __CPROVER_frees(arr)
```

Comment:
What if I call `__CPROVER_frees` with a function returning `void`, but with no pointers in it. For example,
```c
void foo()
{
  int x;
}
...
__CPROVER_frees(foo())
```